src/network/clientpackethandler.cpp

```diff
@@ -764,6 +764,39 @@ void Client::handleCommand_ItemDef(NetworkPacket* pkt)
 	m_itemdef_received = true;
 }
 
+
+void Client::handleCommand_Settings(NetworkPacket* pkt)
+{
+	infostream << "Client: Received settings: packet size: "
+			<< pkt->getSize() << std::endl;
+
+	u8 type;
+	u8 flags;
+	(*pkt) >> type;
+	assert(type == 0);
+	(*pkt) >> flags;
+	bool replace = (flags & 0x01) != 0;
+	if (replace)
+		m_settings_received = true;
+
+	// If managed by server, the settings have already been applied.
+	// No need to override (or actually: overriding is harmful)
+	if (g_settings_managed_by_server)
+		return;
+
+	// Decompress settings
+	std::string datastring(pkt->getString(2), pkt->getSize());
```

Comment:
Re: compression: In the short term, it won't be much indeed, but I assume that over time, the amount of data will grow, and even more so if this is going to be used for client-side scripting. And it may also end up being used for other purposes.
If preferred, I'll remove the compression.

Re: serialization: A value can be a nested group of settings, so the transmission format can't be just a list of `<key,value>` pairs. Also, I suppose a more generic serialization is more future-proof...